extensions/vscode/src/state.ts

```diff
@@ -167,6 +167,25 @@ export class PublisherState implements Disposable {
     }
   }
 
+  updateContentRecord(
+    newValue: ContentRecord | PreContentRecord | PreContentRecordWithConfig,
+  ) {
+    const existingContentRecord = this.findContentRecord(
+      newValue.saveName,
+      newValue.projectDir,
+    );
+    if (existingContentRecord) {
+      const crIndex = this.contentRecords.findIndex(
+        (contentRecord) => contentRecord.id === existingContentRecord?.id,
```

Comment:
Should we be matching on `id` here? If we are updating a pre-deployment won't this check not give us any index and push a new content record to the array?

Actually since we are using this on the failure only we are nearly guaranteed to have an `id`, however since the method is for general use in the state it feels like it should be matching on the other unique information - `saveName` and `projectDir` like the `findContentRecord` does.

Reply:
Good point. I'll update it to use the same criteria as `findContentRecord` does or possibly even simply call it.